app/callbacks.py

```diff
@@ -198,54 +250,56 @@ def display_blocks(
-                    value="GCF_ID",
-                    id={"type": "gm-dropdown-menu", "index": new_block_id},
-                    clearable=False,
-                ),
-                span=6,
-            ),
-            dmc.GridCol(
-                [
-                    dmc.TextInput(
-                        id={"type": "gm-dropdown-ids-text-input", "index": new_block_id},
-                        placeholder="1, 2, 3, ...",
-                        className="custom-textinput",
+    if len(blocks_id) > 1:
+        new_block_id = blocks_id[-1]
+
+        new_block = dmc.Grid(
+            id={"type": "gm-block", "index": new_block_id},
+            children=[
+                dmc.GridCol(
+                    dbc.Button(
+                        [html.I(className="fas fa-plus")],
+                        id={"type": "gm-add-button", "index": new_block_id},
+                        className="btn-primary",
                     ),
+                    span=1,
+                ),
+                dmc.GridCol(
                     dcc.Dropdown(
-                        id={"type": "gm-dropdown-bgc-class-dropdown", "index": new_block_id},
-                        options=GM_DROPDOWN_BGC_CLASS_OPTIONS,
-                        multi=True,
-                        style={"display": "none"},
+                        options=GM_DROPDOWN_MENU_OPTIONS,
+                        value="GCF_ID",
+                        id={"type": "gm-dropdown-menu", "index": new_block_id},
+                        clearable=False,
                     ),
-                ],
-                span=5,
-            ),
-        ],
-        gutter="md",
-    )
+                    span=6,
+                ),
+                dmc.GridCol(
+                    [
+                        dmc.TextInput(
+                            id={"type": "gm-dropdown-ids-text-input", "index": new_block_id},
+                            placeholder="1, 2, 3, ...",
+                            className="custom-textinput",
+                        ),
+                        dcc.Dropdown(
+                            id={"type": "gm-dropdown-bgc-class-dropdown", "index": new_block_id},
+                            options=GM_DROPDOWN_BGC_CLASS_OPTIONS,
+                            multi=True,
+                            style={"display": "none"},
+                        ),
+                    ],
+                    span=5,
+                ),
+            ],
+            gutter="md",
+        )
 
-    # Hide the add button on the previous last block
-    existing_blocks[-1]["props"]["children"][0]["props"]["children"]["props"]["style"] = {
-        "display": "none"
-    }
+        # Hide the add button on the previous last block
+        existing_blocks[-1]["props"]["children"][0]["props"]["children"]["props"]["style"] = {
+            "display": "none"
+        }
 
-    return existing_blocks + [new_block]
+        return existing_blocks + [new_block]
```

Comment:
It looks the type of `existing_blocks` is not compatible with `[new_block]`.

Reply:
Still? How do you see that?

Reply:
Enable pylance type checking in vscode.